torch/_inductor/codegen/cpp.py

```diff
@@ -3858,25 +4023,124 @@ def wrap_fn(fn):
     def codegen_loops(self, code, worksharing):
         self.codegen_loops_impl(self.loop_nest, code, worksharing)
 
+    def update_stores_with_parallel_reduction(self):
+        for kernel in self.kernels:
+            kernel.update_stores_with_parallel_reduction()
+
+    def gen_body(self, code: Optional[BracesBuffer] = None):
+        assert code is not None
+        if_prefix = "C10_LIKELY"
+        for kernel in self.kernels:
+            with contextlib.ExitStack() as stack:
+                if kernel.codegen_conditions(code, if_prefix):
+                    if_prefix = "C10_UNLIKELY"
+                    stack.enter_context(code.indent())
+                    body = kernel.gen_body()
+                    if kernel.inner_itervars:
+                        for idx in kernel.inner_itervars:
+                            start, end = kernel.active_ranges[idx]
+                            body = move_code_under_inner_loop(
+                                body, idx, f"{idx}_tail", start, end
+                            )
+                    code.splice(body)
+
+    def aggregate_reduction_buffers(
+        self, inner_loop_reduction_outer_not: bool, outer_loop: Optional["LoopLevel"]
+    ):
+        # CppKernel/CppVecKernel/CppTile2dKernel have reduction buffers themselves,
+        # here we decide how to aggregate them together and put new reduction buffers
+        # under CppKernelProxy
+        def get_buffer_from_main_loop_kernel(attr: str):
```

Comment:
a bit weird that a getter function doesn't return anything...

Reply:
Yes, this function is actually is **set** the buffer from main to to CppKernelProxy, so I renamed it  `set_buffer_from_main_loop_kernel`